### DISCUSSION TITLE
rewrite headers for matomo

### DISCUSSION THREAD

**Discussion by redpimpernel** [gravitational/teleport#38722] [Q&A]
Hi, I have been using the ssh side of teleport for a bit now, and I have been experimenting with putting some docker web apps behind teleport.

All was going well with some very simple apps, then I tried to set up Motomo. I can successfully get to Matomo though teleport, however, when I go to log in to Matomo (not teleport), I get an error about an invalid "Referer" header.

Matomo error:
```
Error: The form security failed because of an invalid "Referer" header. 
If you are using a proxy server, you must configure Matomo to accept the 
proxy header that forwards the host header. Also, check that your "Referer" 
header is sent correctly. If you previously connected using HTTPS, please 
ensure you are connecting over a secure (SSL/TLS) connection and try again.
```

I have not done anything with http headers before and despite my googling, I am not understanding the issue.

I have followed the steps that are part of the Matomo FAQ: [matomo faq](https://matomo.org/faq/how-to-install/faq_98)

Any guidance on what header to rewrite in teleport and with what value, would be very VERY much appreciated. If someone could also spend a few minutes explaining what the issue actually is, that would be amazing. Looking at http headers makes me feel like when I first started learning how to do calculations on a subnet!!

Thank you all in advance.

**Comment by webvictim** [comment]
Sorry for the delay in response here. In general if you're seeing errors like this, adding `Origin` and `Host` headers by following this step in the troubleshooting guidelines often helps: https://goteleport.com/docs/application-access/troubleshooting-apps/#solution
